### PR TITLE
install-deps.sh: use DTS on centos if GCC is too old

### PR DESCRIPTION
**Pull request by tchaikov**
Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
@shinobu-x could you help test this? the goal here is to make sure that `run-make-check.sh` builds even if running on centos.

**Comment by shinobu-x** [comment]
@tchaikov ack

**Comment by shinobu-x** [comment]
@tchaikov 
```
@octopus:ceph [sk-test-19398]$ rm -fRv build
@octopus:ceph [sk-test-19398]$ ./run-make-check.sh 
./run-make-check.sh: line 72: cd: build: No such file or directory
make: *** No rule to make target `tests'.  Stop.
@octopus:ceph [sk-test-19398]$ ls | grep build | wc -l
1
```

**Comment by shinobu-x** [comment]
```
@octopus:ceph [sk-test-19398]$ ./install-deps.sh 
Using yum to install dependencies
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirrors.aliyun.com
 * epel: mirror.lzu.edu.cn
 * extras: mirrors.aliyun.com
 * updates: mirrors.aliyun.com
Package redhat-lsb-core-4.1-27.el7.centos.1.x86_64 already installed and latest version
Nothing to do
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirrors.aliyun.com
 * epel: mirror.lzu.edu.cn
 * extras: mirrors.aliyun.com
 * updates: mirrors.aliyun.com
Package yum-utils-1.1.31-42.el7.noarch already installed and latest version
Nothing to do
Loaded plugins: fastestmirror
adding repo from: https://dl.fedoraproject.org/pub/epel/7/x86_64/

[dl.fedoraproject.org_pub_epel_7_x86_64_]
name=added from: https://dl.fedoraproject.org/pub/epel/7/x86_64/
baseurl=https://dl.fedoraproject.org/pub/epel/7/x86_64/
enabled=1


Loaded plugins: fastestmirror
dl.fedoraproject.org_pub_epel_7_x86_64_                  | 4.7 kB     00:00     
Loading mirror speeds from cached hostfile
 * base: mirrors.aliyun.com
 * epel: mirror.lzu.edu.cn
 * extras: mirrors.aliyun.com
 * updates: mirrors.aliyun.com
Package epel-release-7-11.noarch already installed and latest version
Nothing to do
Loaded plugins: fastestmirror
=================================== repo: cr ===================================
[cr]
async = True
bandwidth = 0
base_persistdir = /var/lib/yum/repos/x86_64/7
baseurl = http://mirror.centos.org/centos/7/cr/x86_64/
cache = 0
cachedir = /var/cache/yum/x86_64/7/cr
check_config_file_age = True
compare_providers_priority = 80
cost = 1000
deltarpm_metadata_percentage = 100
deltarpm_percentage = 
enabled = True
enablegroups = True
exclude = 
failovermethod = priority
ftp_disable_epsv = False
gpgcadir = /var/lib/yum/repos/x86_64/7/cr/gpgcadir
gpgcakey = 
gpgcheck = True
gpgdir = /var/lib/yum/repos/x86_64/7/cr/gpgdir
gpgkey = file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7
hdrdir = /var/cache/yum/x86_64/7/cr/headers
http_caching = all
includepkgs = 
ip_resolve = 
keepalive = True
keepcache = False
mddownloadpolicy = sqlite
mdpolicy = group:small
mediaid = 
metadata_expire = 21600
metadata_expire_filter = read-only:present
metalink = 
minrate = 0
mirrorlist = 
mirrorlist_expire = 86400
name = CentOS-7 - cr
old_base_cache_dir = 
password = 
persistdir = /var/lib/yum/repos/x86_64/7/cr
pkgdir = /var/cache/yum/x86_64/7/cr/packages
proxy = False
proxy_dict = 
proxy_password = 
proxy_username = 
repo_gpgcheck = False
retries = 10
skip_if_unavailable = False
ssl_check_cert_permissions = True
sslcacert = 
sslclientcert = 
sslclientkey = 
sslverify = True
throttle = 0
timeout = 30.0
ui_id = cr/7/x86_64
ui_repoid_vars = releasever,
   basearch
username = 

Loaded plugins: fastestmirror
cr                                                       | 3.3 kB     00:00     
Loading mirror speeds from cached hostfile
 * base: mirrors.aliyun.com
 * epel: mirror.lzu.edu.cn
 * extras: mirrors.aliyun.com
 * updates: mirrors.aliyun.com
Package centos-release-scl-2-2.el7.centos.noarch already installed and latest version
Nothing to do
Loaded plugins: fastestmirror
Enabling base-source repository
Enabling centos-sclo-rh-source repository
Enabling centos-sclo-sclo-source repository
Enabling epel-source repository
Enabling extras-source repository
Enabling updates-source repository
Loading mirror speeds from cached hostfile
 * base: mirrors.aliyun.com
 * epel: mirror.lzu.edu.cn
 * epel-source: mirror.lzu.edu.cn
 * extras: mirrors.aliyun.com
 * updates: mirrors.aliyun.com
Checking for new repos for mirrors
Getting requirements for /tmp/install-deps.119410/ceph.spec
 --> Already installed : 1:java-1.8.0-openjdk-devel-1.8.0.151-1.b12.el7_4.x86_64
 --> Already installed : sharutils-4.13.3-8.el7.x86_64
 --> Already installed : checkpolicy-2.5-4.el7.x86_64
 --> Already installed : selinux-policy-devel-3.13.1-166.el7_4.5.noarch
 --> Already installed : python-cherrypy-3.2.2-4.el7.noarch
 --> Already installed : python-werkzeug-0.9.1-2.el7.noarch
 --> Already installed : python-pecan-0.4.5-2.el7.noarch
 --> Already installed : socat-1.7.3.2-2.el7.x86_64
 --> Already installed : bc-1.06.95-13.el7.x86_64
 --> Already installed : gperf-3.0.4-8.el7.x86_64
 --> Already installed : cmake-2.8.12.2-2.el7.x86_64
 --> Already installed : cryptsetup-1.7.4-3.el7.x86_64
 --> Already installed : fuse-devel-2.9.2-8.el7.x86_64
 --> Already installed : devtoolset-7-gcc-c++-7.2.1-1.el7.x86_64
 --> Already installed : gdbm-1.10-8.el7.x86_64
 --> Already installed : gperftools-devel-2.4-8.el7.x86_64
 --> Already installed : jq-1.5-1.el7.x86_64
 --> Already installed : leveldb-devel-1.12.0-11.el7.x86_64
 --> Already installed : libaio-devel-0.3.109-13.el7.x86_64
 --> Already installed : libblkid-devel-2.23.2-43.el7.x86_64
 --> Already installed : libcurl-devel-7.29.0-42.el7_4.1.x86_64
 --> Already installed : systemd-devel-219-42.el7_4.4.x86_64
 --> Already installed : libtool-2.4.2-20.el7.x86_64
 --> Already installed : libxml2-devel-2.9.1-6.el7_2.3.x86_64
 --> Already installed : libuuid-devel-2.23.2-43.el7.x86_64
 --> Already installed : 1:make-3.82-21.el7.x86_64
 --> Already installed : parted-3.1-28.el7.x86_64
 --> Already installed : 4:perl-5.16.3-286.el7.x86_64
 --> Already installed : 1:pkgconfig-0.27.1-4.el7.x86_64
 --> Already installed : python-2.7.5-58.el7.x86_64
 --> Already installed : python-devel-2.7.5-58.el7.x86_64
 --> Already installed : python-nose-1.3.7-1.el7.noarch
 --> Already installed : python-requests-2.6.0-1.el7_1.noarch
 --> Already installed : python-virtualenv-1.10.1-4.el7.noarch
 --> Already installed : snappy-devel-1.1.0-3.el7.x86_64
 --> Already installed : systemd-219-42.el7_4.4.x86_64
 --> Already installed : util-linux-2.23.2-43.el7.x86_64
 --> Already installed : 1:valgrind-devel-3.12.0-9.el7_4.x86_64
 --> Already installed : which-2.20-7.el7.x86_64
 --> Already installed : xfsprogs-4.5.0-12.el7.x86_64
 --> Already installed : xfsprogs-devel-4.5.0-12.el7.x86_64
 --> Already installed : xmlstarlet-1.6.1-1.el7.x86_64
 --> Already installed : yasm-1.2.0-4.el7.x86_64
 --> Already installed : boost-random-1.53.0-27.el7.x86_64
 --> Already installed : btrfs-progs-3.19.1-1.el7.x86_64
 --> Already installed : nss-devel-3.28.4-15.el7_4.x86_64
 --> Already installed : keyutils-libs-devel-1.5.8-3.el7.x86_64
 --> Already installed : rdma-core-devel-13-7.el7.x86_64
 --> Already installed : openldap-devel-2.4.44-5.el7.x86_64
 --> Already installed : 1:openssl-devel-1.0.2k-8.el7.x86_64
 --> Already installed : CUnit-devel-2.1.3-13.el7.x86_64
 --> Already installed : redhat-lsb-core-4.1-27.el7.centos.1.x86_64
 --> Already installed : Cython-0.19-5.el7.x86_64
 --> Already installed : python-prettytable-0.7.2-3.el7.noarch
 --> Already installed : python-sphinx-1.1.3-11.el7.noarch
 --> Already installed : python34-devel-3.4.5-4.el7.x86_64
 --> Already installed : python34-setuptools-19.2-3.el7.noarch
 --> Already installed : python34-Cython-0.23.5-1.el7.x86_64
 --> Already installed : lttng-ust-devel-2.4.1-4.el7.x86_64
 --> Already installed : libbabeltrace-devel-1.2.4-3.el7.x86_64
 --> Already installed : expat-devel-2.1.0-10.el7_3.x86_64
 --> Already installed : redhat-rpm-config-9.1.0-76.el7.centos.noarch
 --> Already installed : junit-4.11-8.el7.noarch
No uninstalled build requires
```

**Comment by shinobu-x** [comment]
```
@octopus:ceph [sk-test-19398]$ ./run-make-check.sh 
For best results, run this script as a normal user configured
with the ability to run commands as root via sudo.
Checking hostname sanity... OK
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirrors.aliyun.com
 * epel: mirror.lzu.edu.cn
 * extras: mirrors.aliyun.com
 * updates: mirrors.aliyun.com
Package ccache-3.3.4-1.el7.x86_64 already installed and latest version
Package jq-1.5-1.el7.x86_64 already installed and latest version
Package which-2.20-7.el7.x86_64 already installed and latest version
Nothing to do
Using yum to install dependencies
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirrors.aliyun.com
 * epel: mirror.lzu.edu.cn
 * extras: mirrors.aliyun.com
 * updates: mirrors.aliyun.com
Package redhat-lsb-core-4.1-27.el7.centos.1.x86_64 already installed and latest version
Nothing to do
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirrors.aliyun.com
 * epel: mirror.lzu.edu.cn
 * extras: mirrors.aliyun.com
 * updates: mirrors.aliyun.com
Package yum-utils-1.1.31-42.el7.noarch already installed and latest version
Nothing to do
Loaded plugins: fastestmirror
adding repo from: https://dl.fedoraproject.org/pub/epel/7/x86_64/

[dl.fedoraproject.org_pub_epel_7_x86_64_]
name=added from: https://dl.fedoraproject.org/pub/epel/7/x86_64/
baseurl=https://dl.fedoraproject.org/pub/epel/7/x86_64/
enabled=1


Loaded plugins: fastestmirror
dl.fedoraproject.org_pub_epel_7_x86_64_                  | 4.7 kB     00:00     
Loading mirror speeds from cached hostfile
 * base: mirrors.aliyun.com
 * epel: mirror.lzu.edu.cn
 * extras: mirrors.aliyun.com
 * updates: mirrors.aliyun.com
Package epel-release-7-11.noarch already installed and latest version
Nothing to do
Loaded plugins: fastestmirror
=================================== repo: cr ===================================
[cr]
async = True
bandwidth = 0
base_persistdir = /var/lib/yum/repos/x86_64/7
baseurl = http://mirror.centos.org/centos/7/cr/x86_64/
cache = 0
cachedir = /var/cache/yum/x86_64/7/cr
check_config_file_age = True
compare_providers_priority = 80
cost = 1000
deltarpm_metadata_percentage = 100
deltarpm_percentage = 
enabled = True
enablegroups = True
exclude = 
failovermethod = priority
ftp_disable_epsv = False
gpgcadir = /var/lib/yum/repos/x86_64/7/cr/gpgcadir
gpgcakey = 
gpgcheck = True
gpgdir = /var/lib/yum/repos/x86_64/7/cr/gpgdir
gpgkey = file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7
hdrdir = /var/cache/yum/x86_64/7/cr/headers
http_caching = all
includepkgs = 
ip_resolve = 
keepalive = True
keepcache = False
mddownloadpolicy = sqlite
mdpolicy = group:small
mediaid = 
metadata_expire = 21600
metadata_expire_filter = read-only:present
metalink = 
minrate = 0
mirrorlist = 
mirrorlist_expire = 86400
name = CentOS-7 - cr
old_base_cache_dir = 
password = 
persistdir = /var/lib/yum/repos/x86_64/7/cr
pkgdir = /var/cache/yum/x86_64/7/cr/packages
proxy = False
proxy_dict = 
proxy_password = 
proxy_username = 
repo_gpgcheck = False
retries = 10
skip_if_unavailable = False
ssl_check_cert_permissions = True
sslcacert = 
sslclientcert = 
sslclientkey = 
sslverify = True
throttle = 0
timeout = 30.0
ui_id = cr/7/x86_64
ui_repoid_vars = releasever,
   basearch
username = 

Loaded plugins: fastestmirror
cr                                                       | 3.3 kB     00:00     
Loading mirror speeds from cached hostfile
 * base: mirrors.aliyun.com
 * epel: mirror.lzu.edu.cn
 * extras: mirrors.aliyun.com
 * updates: mirrors.aliyun.com
Package centos-release-scl-2-2.el7.centos.noarch already installed and latest version
Nothing to do
Loaded plugins: fastestmirror
Enabling base-source repository
Enabling centos-sclo-rh-source repository
Enabling centos-sclo-sclo-source repository
Enabling epel-source repository
Enabling extras-source repository
Enabling updates-source repository
Loading mirror speeds from cached hostfile
 * base: mirrors.aliyun.com
 * epel: mirror.lzu.edu.cn
 * epel-source: mirror.lzu.edu.cn
 * extras: mirrors.aliyun.com
 * updates: mirrors.aliyun.com
Checking for new repos for mirrors
Getting requirements for /tmp/install-deps.119786/ceph.spec
 --> Already installed : 1:java-1.8.0-openjdk-devel-1.8.0.151-1.b12.el7_4.x86_64
 --> Already installed : sharutils-4.13.3-8.el7.x86_64
 --> Already installed : checkpolicy-2.5-4.el7.x86_64
 --> Already installed : selinux-policy-devel-3.13.1-166.el7_4.5.noarch
 --> Already installed : python-cherrypy-3.2.2-4.el7.noarch
 --> Already installed : python-werkzeug-0.9.1-2.el7.noarch
 --> Already installed : python-pecan-0.4.5-2.el7.noarch
 --> Already installed : socat-1.7.3.2-2.el7.x86_64
 --> Already installed : bc-1.06.95-13.el7.x86_64
 --> Already installed : gperf-3.0.4-8.el7.x86_64
 --> Already installed : cmake-2.8.12.2-2.el7.x86_64
 --> Already installed : cryptsetup-1.7.4-3.el7.x86_64
 --> Already installed : fuse-devel-2.9.2-8.el7.x86_64
 --> Already installed : devtoolset-7-gcc-c++-7.2.1-1.el7.x86_64
 --> Already installed : gdbm-1.10-8.el7.x86_64
 --> Already installed : gperftools-devel-2.4-8.el7.x86_64
 --> Already installed : jq-1.5-1.el7.x86_64
 --> Already installed : leveldb-devel-1.12.0-11.el7.x86_64
 --> Already installed : libaio-devel-0.3.109-13.el7.x86_64
 --> Already installed : libblkid-devel-2.23.2-43.el7.x86_64
 --> Already installed : libcurl-devel-7.29.0-42.el7_4.1.x86_64
 --> Already installed : systemd-devel-219-42.el7_4.4.x86_64
 --> Already installed : libtool-2.4.2-20.el7.x86_64
 --> Already installed : libxml2-devel-2.9.1-6.el7_2.3.x86_64
 --> Already installed : libuuid-devel-2.23.2-43.el7.x86_64
 --> Already installed : 1:make-3.82-21.el7.x86_64
 --> Already installed : parted-3.1-28.el7.x86_64
 --> Already installed : 4:perl-5.16.3-286.el7.x86_64
 --> Already installed : 1:pkgconfig-0.27.1-4.el7.x86_64
 --> Already installed : python-2.7.5-58.el7.x86_64
 --> Already installed : python-devel-2.7.5-58.el7.x86_64
 --> Already installed : python-nose-1.3.7-1.el7.noarch
 --> Already installed : python-requests-2.6.0-1.el7_1.noarch
 --> Already installed : python-virtualenv-1.10.1-4.el7.noarch
 --> Already installed : snappy-devel-1.1.0-3.el7.x86_64
 --> Already installed : systemd-219-42.el7_4.4.x86_64
 --> Already installed : util-linux-2.23.2-43.el7.x86_64
 --> Already installed : 1:valgrind-devel-3.12.0-9.el7_4.x86_64
 --> Already installed : which-2.20-7.el7.x86_64
 --> Already installed : xfsprogs-4.5.0-12.el7.x86_64
 --> Already installed : xfsprogs-devel-4.5.0-12.el7.x86_64
 --> Already installed : xmlstarlet-1.6.1-1.el7.x86_64
 --> Already installed : yasm-1.2.0-4.el7.x86_64
 --> Already installed : boost-random-1.53.0-27.el7.x86_64
 --> Already installed : btrfs-progs-3.19.1-1.el7.x86_64
 --> Already installed : nss-devel-3.28.4-15.el7_4.x86_64
 --> Already installed : keyutils-libs-devel-1.5.8-3.el7.x86_64
 --> Already installed : rdma-core-devel-13-7.el7.x86_64
 --> Already installed : openldap-devel-2.4.44-5.el7.x86_64
 --> Already installed : 1:openssl-devel-1.0.2k-8.el7.x86_64
 --> Already installed : CUnit-devel-2.1.3-13.el7.x86_64
 --> Already installed : redhat-lsb-core-4.1-27.el7.centos.1.x86_64
 --> Already installed : Cython-0.19-5.el7.x86_64
 --> Already installed : python-prettytable-0.7.2-3.el7.noarch
 --> Already installed : python-sphinx-1.1.3-11.el7.noarch
 --> Already installed : python34-devel-3.4.5-4.el7.x86_64
 --> Already installed : python34-setuptools-19.2-3.el7.noarch
 --> Already installed : python34-Cython-0.23.5-1.el7.x86_64
 --> Already installed : lttng-ust-devel-2.4.1-4.el7.x86_64
 --> Already installed : libbabeltrace-devel-1.2.4-3.el7.x86_64
 --> Already installed : expat-devel-2.1.0-10.el7_3.x86_64
 --> Already installed : redhat-rpm-config-9.1.0-76.el7.centos.noarch
 --> Already installed : junit-4.11-8.el7.noarch
No uninstalled build requires
/usr/bin/ccache
enabling ccache
-- The CXX compiler identification is GNU 4.8.5
-- The C compiler identification is GNU 4.8.5
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Building with ccache: /usr/bin/ccache, CCACHE_DIR=/root/.ccache
-- Looking for fallocate
-- Looking for fallocate - found
-- Looking for posix_fadvise
-- Looking for posix_fadvise - found
-- Looking for posix_fallocate
-- Looking for posix_fallocate - found
-- Looking for syncfs
-- Looking for syncfs - found
-- Looking for sync_file_range
-- Looking for sync_file_range - found
-- Looking for pwritev
-- Looking for pwritev - found
-- Looking for splice
-- Looking for splice - found
-- Looking for getgrouplist
-- Looking for getgrouplist - found
-- Looking for fdatasync
-- Looking for fdatasync - found
-- Looking for strerror_r
-- Looking for strerror_r - found
-- Looking for name_to_handle_at
-- Looking for name_to_handle_at - found
-- Looking for pipe2
-- Looking for pipe2 - found
-- Looking for pthread_spin_init
-- Looking for pthread_spin_init - found
-- Looking for pthread_set_name_np
-- Looking for pthread_set_name_np - not found
-- Looking for pthread_setname_np
-- Looking for pthread_setname_np - found
-- Looking for pthread_getname_np
-- Looking for pthread_getname_np - found
-- Looking for pthread_rwlockattr_setkind_np
-- Looking for pthread_rwlockattr_setkind_np - found
-- Looking for eventfd
-- Looking for eventfd - found
-- Looking for getprogname
-- Looking for getprogname - not found
-- Looking for include file linux/types.h
-- Looking for include file linux/types.h - found
-- Looking for include file linux/version.h
-- Looking for include file linux/version.h - found
-- Looking for include file arpa/nameser_compat.h
-- Looking for include file arpa/nameser_compat.h - found
-- Looking for include file sys/mount.h
-- Looking for include file sys/mount.h - found
-- Looking for include file sys/param.h
-- Looking for include file sys/param.h - found
-- Looking for include file sys/types.h
-- Looking for include file sys/types.h - found
-- Looking for include file sys/vfs.h
-- Looking for include file sys/vfs.h - found
-- Looking for include file sys/prctl.h
-- Looking for include file sys/prctl.h - found
-- Looking for include file execinfo.h
-- Looking for include file execinfo.h - found
-- Looking for include file sched.h
-- Looking for include file sched.h - found
-- Looking for include file valgrind/helgrind.h
-- Looking for include file valgrind/helgrind.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of __be16
-- Check size of __be16 - done
-- Check size of __be32
-- Check size of __be32 - done
-- Check size of __be64
-- Check size of __be64 - done
-- Check size of __le16
-- Check size of __le16 - done
-- Check size of __le32
-- Check size of __le32 - done
-- Check size of __le64
-- Check size of __le64 - done
-- Check size of __u8
-- Check size of __u8 - done
-- Check size of __u16
-- Check size of __u16 - done
-- Check size of __u32
-- Check size of __u32 - done
-- Check size of __u64
-- Check size of __u64 - done
-- Check size of __s8
-- Check size of __s8 - done
-- Check size of __s16
-- Check size of __s16 - done
-- Check size of __s32
-- Check size of __s32 - done
-- Check size of __s64
-- Check size of __s64 - done
-- Looking for res_nquery
-- Looking for res_nquery - found
-- Looking for F_SETPIPE_SZ
-- Looking for F_SETPIPE_SZ - found
-- Looking for __func__
-- Looking for __func__ - found
-- Looking for __PRETTY_FUNCTION__
-- Looking for __PRETTY_FUNCTION__ - found
-- Looking for getentropy
-- Looking for getentropy - not found
-- Performing Test STRERROR_R_CHAR_P
-- Performing Test STRERROR_R_CHAR_P - Success
-- Performing Test HAVE_STAT_ST_MTIM_TV_NSEC
-- Performing Test HAVE_STAT_ST_MTIM_TV_NSEC - Success
-- Performing Test HAVE_STAT_ST_MTIMESPEC_TV_NSEC
-- Performing Test HAVE_STAT_ST_MTIMESPEC_TV_NSEC - Failed
-- Found libibverbs: /usr/lib64/libibverbs.so
-- Performing Test HAVE_IBV_EXP
-- Performing Test HAVE_IBV_EXP - Failed
-- Found ibverbs: /usr/lib64/libibverbs.so  
-- Looking for backtrace
-- Looking for backtrace - found
-- backtrace facility detected in default set of libraries
-- Found Backtrace: /usr/include  
-- Found udev: /usr/lib64/libudev.so  
-- Found blkid: /usr/lib64/libblkid.so  
-- Found OpenLdap: /usr/include  
-- Found fuse: /usr/include/fuse  
-- Found xfs: /usr/lib64/libhandle.so  
-- Found aio: /lib64/libaio.so  
-- Found CUnit: /usr/lib64/libcunit.so  
-- Found uuid: /usr/lib64/libuuid.so  
-- Found leveldb: /lib64/libleveldb.so  
-- Found snappy: /lib64/libsnappy.so  
-- Found gperftools: /lib64/libtcmalloc.so (found version "2.4") 
-- Found keyutils: /usr/lib64/libkeyutils.so  
-- Found CURL: /usr/lib64/libcurl.so (found version "7.29.0") 
-- Looking for curl_multi_wait
-- Looking for curl_multi_wait - found
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.27.1") 
-- checking for module 'nss'
--   found nss, version 3.28.4
-- NSS_LIBRARIES: /usr/lib64/libssl3.so;/usr/lib64/libsmime3.so;/usr/lib64/libnss3.so;/usr/lib64/libnssutil3.so
-- NSS_INCLUDE_DIRS: /usr/include/nss3
-- Found NSS: /usr/lib64/libssl3.so;/usr/lib64/libsmime3.so;/usr/lib64/libnss3.so;/usr/lib64/libnssutil3.so  
-- checking for module 'nspr'
--   found nspr, version 4.13.1
-- Found NSPR: /usr/lib64/libplds4.so;/usr/lib64/libplc4.so;/usr/lib64/libnspr4.so  
-- SSL with NSS selected (Libs: /usr/lib64/libssl3.so;/usr/lib64/libsmime3.so;/usr/lib64/libnss3.so;/usr/lib64/libnssutil3.so)
-- Found EXPAT: /usr/lib64/libexpat.so (found version "2.1.0") 
-- Looking for openssl anyways, because radosgw selected
-- Found OpenSSL: /usr/lib64/libssl.so;/usr/lib64/libcrypto.so (found version "1.0.2k") 
-- ssl soname: libssl.so.10
-- crypto soname: libcrypto.so.10
-- Found PythonInterp: /usr/bin/python2.7 (found suitable version "2.7.5", minimum required is "2.7") 
-- Found PythonLibs: /usr/lib64/libpython2.7.so (found suitable version "2.7.5", minimum required is "2.7") 
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.7") 
-- Found LTTngUST: /usr/lib64/liblttng-ust.so;dl (found version "2.4.1") 
-- Found babeltrace: /usr/include  
-- BUILDING Boost Libraries at j 16
-- boost will be downloaded...
-- Found Boost: /git/ceph/build/boost/include  
-- Performing Test HAS_FORTIFY_SOURCE
-- Performing Test HAS_FORTIFY_SOURCE - Success
-- Performing Test HAS_STACK_PROTECT
-- Performing Test HAS_STACK_PROTECT - Success
-- Performing Test HAVE_INTEL_SSE
-- Performing Test HAVE_INTEL_SSE - Success
-- Performing Test HAVE_INTEL_SSE2
-- Performing Test HAVE_INTEL_SSE2 - Success
-- Performing Test HAVE_INTEL_SSE3
-- Performing Test HAVE_INTEL_SSE3 - Success
-- Performing Test HAVE_INTEL_SSSE3
-- Performing Test HAVE_INTEL_SSSE3 - Success
-- Performing Test HAVE_INTEL_PCLMUL
-- Performing Test HAVE_INTEL_PCLMUL - Success
-- Performing Test HAVE_INTEL_SSE4_1
-- Performing Test HAVE_INTEL_SSE4_1 - Success
-- Performing Test HAVE_INTEL_SSE4_2
-- Performing Test HAVE_INTEL_SSE4_2 - Success
--  we have a modern and working yasm
--  we are x84_64
-- Performing Test not_arch_x32
-- Performing Test not_arch_x32 - Success
--  we are not x32
--  yasm can also build the isa-l stuff
-- Performing Test COMPILER_SUPPORTS_CXX11
-- Performing Test COMPILER_SUPPORTS_CXX11 - Success
-- Performing Test COMPILER_SUPPORTS_GNU99
-- Performing Test COMPILER_SUPPORTS_GNU99 - Success
CMake Error at src/CMakeLists.txt:176 (message):
  performance regression is expected due to an O(n) implementation of
  'std::list::size()' in libstdc++ older than 5.1.0, Please use GCC 5.1 and
  up.


-- Performing Test COMPILER_SUPPORTS_DIAGNOSTICS_COLOR
-- Performing Test COMPILER_SUPPORTS_DIAGNOSTICS_COLOR - Success
-- Found PythonInterp: /usr/bin/python2.7 (found suitable version "2.7.5", minimum required is "2") 
-- Found PythonLibs: /usr/lib64/libpython2.7.so (found suitable version "2.7.5", minimum required is "2") 
-- Performing Test HAS_VTA
-- Performing Test HAS_VTA - Success
--  Using EventEpoll for events.
-- MergeStaticLibraries: cephd_cls: merged cephd_cls_base;liblua
-- MergeStaticLibraries: cephd_cls: remaining m;/usr/lib64/libdl.so
-- Found cython
-- Check if the system is big endian
-- Searching 16 bit integer
-- Check size of unsigned short
-- Check size of unsigned short - done
-- Using unsigned short
-- Check if the system is big endian - little endian
-- MergeStaticLibraries: cephd_ec: merged cephd_ec_base;cephd_ec_isa;cephd_ec_jerasure;cephd_ec_lrc;cephd_ec_shec
-- Found PythonInterp: /usr/bin/python2.7 (found version "2.7.5") 
-- MergeStaticLibraries: cephd_compressor: merged cephd_compressor_base;cephd_compressor_snappy;cephd_compressor_zlib;cephd_compressor_zstd
-- MergeStaticLibraries: cephd_rbd: merged cephd_rbd_base;rbd_internal;rbd_types;journal
-- MergeStaticLibraries: cephd_rbd: remaining osdc;cls_journal_client
-- MergeStaticLibraries: cephd_rgw: merged cephd_rgw_base;rgw_a;radosgw_a
-- MergeStaticLibraries: cephd_rgw: remaining librados;cls_lock_client;cls_rgw_client;cls_refcount_client;cls_log_client;cls_statelog_client;cls_timeindex_client;cls_version_client;cls_replica_log_client;cls_user_client;ceph-common;common_utf8;global;/usr/lib64/libcurl.so;/usr/lib64/libexpat.so;/usr/lib64/libldap.so;/usr/lib64/liblber.so;/usr/lib64/libssl3.so;/usr/lib64/libsmime3.so;/usr/lib64/libnss3.so;/usr/lib64/libnssutil3.so;/usr/lib64/libplds4.so;/usr/lib64/libplc4.so;/usr/lib64/libnspr4.so
-- MergeStaticLibraries: cephd: merged cephd_base;cephd_compressor;cephd_ec;cephd_cls;cephd_cls_kvs;cephd_rados;common;common_utf8;erasure_code;global;json_spirit;kv;mds;mon;os;osd;osdc;rocksdb;cephd_rgw;cephd_rbd
-- MergeStaticLibraries: cephd: remaining m;/usr/lib64/libdl.so;resolv;Boost::thread;Boost::system;Boost::regex;Boost::random;Boost::program_options;Boost::date_time;Boost::iostreams;/usr/lib64/libblkid.so;/usr/lib64/libssl3.so;/usr/lib64/libsmime3.so;/usr/lib64/libnss3.so;/usr/lib64/libnssutil3.so;/usr/lib64/libplds4.so;/usr/lib64/libplc4.so;/usr/lib64/libnspr4.so;-lpthread;dl;/usr/lib64/libibverbs.so;/usr/lib64/libz.so;ceph-common;dpdk::pci;dpdk::bus_pci;dpdk::eal;dpdk::mempool;dpdk::mempool_ring;dpdk::mempool_stack;dpdk::ring;rt;pthread;/lib64/libleveldb.so;/lib64/libtcmalloc.so;/lib64/libsnappy.so;liblua;/lib64/libaio.so;/usr/lib64/libfuse.so;spdk::nvme;spdk::log;spdk::lvol;spdk::env_dpdk;spdk::util;dmclock;librados;cls_lock_client;cls_rgw_client;cls_refcount_client;cls_log_client;cls_statelog_client;cls_timeindex_client;cls_version_client;cls_replica_log_client;cls_user_client;/usr/lib64/libcurl.so;/usr/lib64/libexpat.so;/usr/lib64/libldap.so;/usr/lib64/liblber.so;cls_journal_client
-- exclude following files under src: *.js;*.css;civetweb;erasure-code/jerasure/jerasure;erasure-code/jerasure/gf-complete;rocksdb;googletest;spdk;xxHash;isa-l;lua;Beast;dpdk;zstd;crypto/isa-l/isa-l_crypto;blkin;rapidjson
-- Configuring incomplete, errors occurred!
See also "/git/ceph/build/CMakeFiles/CMakeOutput.log".
See also "/git/ceph/build/CMakeFiles/CMakeError.log".
done.
./run-make-check.sh: line 72: cd: build: No such file or directory
make: *** No rule to make target `tests'.  Stop.

```

**Comment by shinobu-x** [comment]
```
@octopus:ceph [sk-test-19398]$ ./do_cmake.sh 
+ git submodule update --init --recursive
+ test -e build
+ echo 'build dir already exists; rm -rf build and re-run'
build dir already exists; rm -rf build and re-run
+ exit 1
```

**Comment by tchaikov** [comment]
@shinobu-x thanks for testing. fixed and repushed.

**Comment by liewegas** [comment]
lgtm!

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-install-deps-use-dts/b38621f30258fd7b45e940bcd2b1b6ee25f86be5/

File: install-deps.sh [hunk 1]
```diff
@@ -50,6 +50,38 @@ function ensure_decent_gcc {
     $SUDO ln -nsf /usr/bin/g++ /usr/bin/x86_64-linux-gnu-g++
 }
 
+function version_lt {
+    test $1 != $(echo -e "$1\n$2" | sort -rV | head -n 1)
```
Comment (smithfarm) [inline install-deps.sh]: Nice!

**Pull request merged**